src/task.js

```diff
@@ -0,0 +1,110 @@
+// This a class containing all the necessary operations
+import dots from './images/menu.png';
+import unchecked from './images/unchecked.png';
+import mark from './images/check.png';
+import recycle from './images/bin.png';
+
+const ul = document.getElementById('to-do-list');
+export default class TaskHandling {
+  constructor(tasks) {
+    this.tasks = tasks;
+    this.taskCount = tasks.length;
+    ul.innerHTML = '';
+    this.populate();
+  }
+
+  add(ele) {
+    if (ele.value === '') {
+      return;
+    }
+    this.taskCount += 1;
+    this.tasks.push({ index: this.taskCount, desc: ele.value, completed: false });
+    localStorage.setItem('listOfTasks', JSON.stringify(this.tasks));
+    ul.innerHTML = '';
+    ele.value = '';
+    this.populate();
+  }
+
+  remove(ele) {
+    this.tasks = this.tasks.filter((e) => e.index !== +ele.parentElement.id);
+    for (let i = 0; i < this.tasks.length; i += 1) {
+      this.tasks[i].index = i + 1;
+    }
+    localStorage.setItem('listOfTasks', JSON.stringify(this.tasks));
+    ul.removeChild(ele.parentElement);
+  }
+
+  edit(ele) {
+    ele.classList.add('hide');
+    const li = ele.parentElement;
+    const textArea = ele.nextSibling;
+    const menu = textArea.nextSibling;
+    const bin = menu.nextSibling;
+
+    textArea.classList.remove('hide');
+    menu.classList.add('hide');
+    bin.classList.remove('hide');
+    li.classList.add('edit');
+
+    textArea.focus();
+    textArea.onblur = () => {
+      if (textArea.value === '') {
+        this.remove(textArea);
+        return;
+      }
+
+      ele.textContent = textArea.value;
+      menu.classList.remove('hide');
+      bin.classList.add('hide');
+      textArea.classList.add('hide');
+      ele.classList.remove('hide');
+      li.classList.remove('edit');
+    };
+  }
```

Comment:
- [ ] Kindly make sure that when a task is edited, the editing persist and the storage is updated

Reply:
Yeah, Forget to update the local storage. Thank you sir